spring-integration-core/src/main/java/org/springframework/integration/store/SimpleMessageStore.java

```diff
@@ -233,8 +270,8 @@ public void removeMessageGroup(Object groupId) {
 					return;
 				}
 
-				groupUpperBound.release(groupIdToMessageGroup.get(groupId).size());
-				groupIdToMessageGroup.remove(groupId);
+				this.groupToUpperBound.remove(groupId).release(this.groupCapacity);
```

Comment:
I don't think so. Because the code here is enough robust from the concurrent access:
```
lock.lockInterruptibly();
try {
        if (!groupIdToMessageGroup.containsKey(groupId)) {
	    return;
	}
	this.groupToUpperBound.remove(groupId).release(this.groupCapacity);
	this.groupIdToMessageGroup.remove(groupId);
}
finally {
	lock.unlock();
}
```
You can see the `lock` and a couple of modification operators if `groupId` exists.